src/ServerScriptService/Services/WeldService.luau

```diff
@@ -0,0 +1,68 @@
+local ReplicatedStorage = game:GetService("ReplicatedStorage")
+local Graph = require(ReplicatedStorage.Common.Graph)
+local Knit = require(ReplicatedStorage.Packages.Knit)
+
+local WeldService = Knit.CreateService {
+    Name = "WeldService";
+    Client = {};
+}
+
+function WeldService:WeldModel(model)
+    local primaryPart = assert(model.PrimaryPart, "Model has no primary part!")
+
+    for _, v in model:GetChildren() do
+        if v == model.PrimaryPart then
+            continue
+        end
+
+        local weld = Instance.new("WeldConstraint")
+        weld.Part0 = primaryPart
+        weld.Part1 = v
+        weld.Parent = model
```

Comment:
There's a bit of code duplication here. A private method for creating welds (part1, part2, parent) would be nice.